src/redux/books/booksSlice.js

```diff
@@ -1,13 +1,41 @@
 import { createSlice } from '@reduxjs/toolkit';
 
-const booksSlice = createSlice({
+const initialState = [
+  {
+    item_id: 'item1',
+    title: 'The Great Gatsby',
+    author: 'John Smith',
+    category: 'Fiction',
+  },
+  {
+    item_id: 'item2',
+    title: 'Anna Karenina',
+    author: 'Leo Tolstoy',
+    category: 'Fiction',
+  },
+  {
+    item_id: 'item3',
+    title: 'The Selfish Gene',
+    author: 'Richard Dawkins',
+    category: 'Nonfiction',
+  },
+];
+
+export const booksSlice = createSlice({
   name: 'books',
-  initialState: [],
+  initialState,
   reducers: {
     addBook: (state, action) => {
+      // Add a book to the state
       state.push(action.payload);
     },
-    removeBook: (state, action) => state.filter((book) => book.id !== action.payload),
+    removeBook: (state, action) => {
+      // Remove a book from the state by id
+      const bookIndex = state.findIndex((book) => book.item_id === action.payload);
+      if (bookIndex !== -1) {
+        state.splice(bookIndex, 1);
```

Comment:
- [ ]  For the remove book action, kindly use `Array.filter()` method for that, as `splice()` mutates the content of the array which we do not want.

You can try this line of code out. :point_down: 

```JS
removeBook: (state, action) => {
      state.books = state.books.filter((book) => book.id !== action.id)
    },
```